ambry-router/src/main/java/com.github.ambry.router/DeleteOperation.java

```diff
@@ -139,19 +147,32 @@ private DeleteRequest createDeleteRequest() {
    * that different {@link ServerErrorCode} are received from different replicas. These error codes
    * are eventually resolved to a single {@link RouterErrorCode}.
    * @param responseInfo The response to be handled.
+   * @param shouldOnlyRecordMetrics {@code true} indicates that only update metrics for the corresponding
+   *                                            metrics, and do not handle the response.
    */
-  void handleResponse(ResponseInfo responseInfo) {
+  void handleResponse(ResponseInfo responseInfo, boolean shouldOnlyRecordMetrics) {
```

Comment:
This does not look like a good change. Would rather not have such things pollute the main code.